File: ros/ws02/first_ws/src/use_custom2_msg_pkg/scripts/publisher_cus2_msg.py
#! /usr/bin/env python
# coding:utf8

import rospy
from custom2_msgs.msg import Team
from custom_msgs.msg import Student

if __name__ == '__main__':

    nodeName = "publisher_node"
    rospy.init_node(nodeName)

    topicName = "/zcb01/topic"
    pub = rospy.Publisher(topicName, Team, queue_size=1000)

    rate = rospy.Rate(2)
    msg = Team()
    stu = Student()
    idx = 0
    while not rospy.is_shutdown():
        msg.name = "NCHU%d" % idx
        stu.name = "tom%d" % idx
        stu.age = idx
        msg.leader = stu
        pub.publish(msg)

        idx += 1
        rate.sleep()

File: ros/ws02/first_ws/src/use_custom2_msg_pkg/src/publisher_cus2_msg.cpp
//
// Created by zcb on 5/3/21.
//

#include <ros/ros.h>
#include <iostream>
#include <custom2_msgs/Team.h>


int main(int argc, char **argv) {

    std::string nodeName = "publisher_node";
    ros::init(argc, argv, nodeName);
    ros::NodeHandle node;

    std::string topicName = "/zcb01/topic";
    ros::Publisher publisher = node.advertise<custom2_msgs::Team>(topicName, 1000);

    custom2_msgs::Team msg;
    custom_msgs::Student leader;

    ros::Rate rate(2);
    int idx = 0;
    while (ros::ok()) {
        msg.name = "NCHU" + std::to_string(idx);
        leader.name = "tom" + std::to_string(idx);
        leader.age = idx;
        msg.leader = leader;

        publisher.publish(msg);

        idx++;
        rate.sleep();
    }

    return 0;
}